R/fixest-tidiers.R

```diff
@@ -0,0 +1,197 @@
+#'   context. The package also supports robust (i.e. White) and clustered
+#'   standard error reporting via the generic `summary.fixest()` command. In a
+#'   similar vein, the `tidy()` method for these models allows users to specify
+#'   a desired standard error correction either 1) implicitly via the supplied
+#'   fixest object, or 2) explicitly as part of the tidy call. See examples
+#'   below.
+#'
+#' @examples
+#' \donttest{
+#' library(fixest)
+#' gravity <- feols(log(Euros) ~ log(dist_km) | Origin + Destination + Product + Year, trade)
+#'
+#' tidy(gravity)
+#' glance(gravity)
+#' augment(gravity, trade)
+#'
+#' ## To get robust or clustered SEs, users can either:
+#' # 1) Feed tidy() a summary.fixest object that has already accepted these arguments
+#' gravity_summ <- summary(gravity, cluster = c("Product", "Year"))
+#' tidy(gravity_summ, conf.int = TRUE)
+#' # 2) Or, specify the arguments directly in the tidy() call
+#' tidy(gravity, conf.int = TRUE, cluster = c("Product", "Year"))
+#' tidy(gravity, conf.int = TRUE, se = "threeway")
+#' # etc.
+#'
+#' ## The other fixest methods all working similarly. For example:
+#' gravity_pois <- feglm(Euros ~ log(dist_km) | Origin + Destination + Product + Year, trade)
+#' tidy(gravity_pois)
+#' glance(gravity_pois)
+#' augment(gravity_pois, trade)
+#' }
+#'
+#' @export
+#' @family fixest tidiers
+#' @seealso [tidy()], [fixest::feglm()], [fixest::fenegbin()],
+#' [fixest::feNmlm()], [fixest::femlm()], [fixest::feols()], [fixest::fepois()]
+tidy.fixest <- function(x, conf.int = FALSE, conf.level = 0.95, ...) {
+  has_dots <- !missing(...)
+  # These two options are necessary because the user might specify
+  # broom(x, cluster="something") or
+  # broom(summary(x, cluster="something"))
+  # and calling summary() again will overwrite the arguments in the second case.
+  # Note that class(summary(x)) == class(x) == "fixest".
+  if (has_dots) {
+    # Option 1) User specifies se or cluster arguments as part of tidy() call
+    coeftable <- summary(x, ...)$coeftable
+  } else {
+    # Option 2) Otherwise, just use whatever object the user feeds into tidy()
+    coeftable <- x$coeftable
+  }
+  nn <- c("estimate", "std.error", "statistic", "p.value")
+  ret <- fix_data_frame(coeftable, nn)
+
+  # CIs (if desired) will follow a simiar pattern as above, depending on what
+  # the user specifies in the tidy() call.
+  if (conf.int) {
+    if (has_dots) {
+      CI <- stats::confint(x, level = conf.level, ...)
+      # Bind to rest of tibble
+      colnames(CI) <- c("conf.low", "conf.high")
+      ret <- cbind(ret, unrowname(CI))
+    } else {
+      # It's possible we were passed a summary of a fixest object. If no `...`
+      # arguments were specified, then we want to use the std error that `x`
+      # already has. That is, we don't want to call confint() because
+      # doing so throws away whatever arguments were provided to summary().
+      ci_fact <- abs(qnorm((1 - conf.level) / 2))
```

Comment:
This assumptions infinity degrees of freedom, correct? Is there somewhere we can get this information? If not, options include:

- Giving an error with a suggestion for the user on how to get a correct CI
- Giving the asymptotic CI, and clearly documenting this

Reply:
Hmm, I'm just borrowing from the `confint.fixest` code: https://github.com/lrberge/fixest/blob/acfb6877633d7fbf75a7571c15238e857c73c9ee/R/MiscFuns.R#L6197-L6202

Reply:
Let's add a note to the documentation stating that CIs assume infinite degrees of freedom then.